## Breast cancer quantitative proteome and proteogenomic landscape 

Johansson et al. 2019

www. breastcancerlandscape.org

### SUMMARY

an unbiased analysis of **breast tumor proteomes**, inclusive of **9995 proteins** quantified across all tumors, for the first time **recapitulates BC subtypes**. 

**poor-prognosis basal-like and luminal B tumors are further subdivided by immune component infiltration**, suggesting the current classification is incomplete

proteome-based **networks** distinguish **functional protein modules** for breast tumor groups, with co-expression of EGFR and MET marking ductal carcinoma in situ regions of normal-like tumors and lending to a **more accurate classification** of this poorly defined subtype



Genes included within **prognostic mRNA panels have significantly higher than average mRNA-protein correlations**, and **gene copy number alterations are dampened at the protein-level**



**protein products mapping to non-coding genomic regions** are identified -> a potential new class of tumor-specific immunotherapeutic targets.



### INTRO

**PAM50** = a set of 50 transcripts (collectively known as PAM50)  established for the five subtypes 

1. basal-like (highly proliferative)
2. HER2 (highly proliferative)
3. luminal A 
4. luminal B (highly proliferative)
5. normal-like

 surrogate immunohistochemistry **(IHC) markers** implemented to partially recapitulate the strati- fying and prognostic information garnered in the original studies. 

BUT

* multigene expression assays are not readily available to all patients, and
* **despite progress in the development of pathology-based surrogate PAM50 markers, one out of three patients are still potentially misclassified**

Though mRNA-based classifica- tions have had great clinical utility, certain shortcomings may be attributable to **varying protein–mRNA abundance** **correlations** and the **inability of mRNA measurements to capture ligand- mediated interplay between tumor and host** and characterize the **extracellular space**

**Proteo- genomics studies** link aberrations observed at the protein level to genomic events or properties; These analyses can identify protein products of genomic regions, previously thought to be silent and/or specifically expressed in transformed tissue that could represent **neoantigens** (immunotherapy !)



in-depth quantitative profile of the proteomes of 45 breast tumors, 9 represented from each of the 5 PAM50-based molecular classifications. 



the first to **recapitulate the current mRNA-based molecular classifi- cations with an unsupervised analysis of whole-proteome data**



Independent layers of analyses 

* reveal **novel immunohistochemical biomarker candidates** to more reliably **stratify difficult-to-classify patients** for treatment options
* provide a proteome-based framework to **assess prognosis** for those straddling treatment class assignments
* link **immune cell infiltration and tumor extracellular matrix compo- sition to prognosis**
* **connect molecular classification to metabolic phenotype** 
* enables application of proteogenomic analyses and the **discovery of neoantigens** arising from tumor-specific variants of known proteins and regions of the genome previously thought to be noncoding



### RESULTS

LC-MS/MS-based protein quantification

13,997 protein products of 12,645 genes 

9995 proteins quantified  in each of the 45 tumors, based on gene symbol centric quantification used for all quantitative proteome analyses 

### Correlation analysis of tumor proteomes and metabolomes

Unsupervised hierarchical clustering of proteome profiles stra- tifies tumors largely in agreement with the PAM50 subtypes

the luminal B and HER2 subtypes are intermixed, indicating similarities in the molecular phenotype.

clustering of relative protein quantities, across tumor samples, groups proteins in accordance with their known co-functions in BC biology

plasma and erythrocyte proteins originate outside of the tumor and would not be detected by transcriptional profiling, demonstrating the unique capability of proteome profiling to consider the tumor in the context of systemic functions of the host 

tumor composition correlates between MS- and histopathology-based evaluation 



Hierarchical clustering of phosphoprotein correlation profiles (RPPA-quantified phosphoprotein abundance to MS-quantified protein abundance) divides phosphoproteins into four distinct groups 

respectively, regulate proteins associated with proliferation, blood plasma, the HER2 amplicon, and the luminal subtype.



PAM50 subtype assignments are based on mRNA profile distance to subtype centroid

High PAM50 subtype agreement with correlation-based hierarchical clustering of tumor protein expression profiles **considering only the 37 PAM50 gene members in the quantified proteome** demonstrates **the patient-stratifying information contained within the entire proteome is derived from a smaller subset**

unsupervised hierarchical clustering of correlations to each PAM50 subtype centroid demonstrates some ambiguity in the classificatio



6 six consensus core tumor clusters (CoTC) : core sets of tumors whose proteomes are representative of a proteome-based grouping defined using unsupervised cluster- ing based on high-variance protein (n = 1334) abundance profiles 

CoTC assignments overlap with PAM50 normal-like and luminal A classifications, but **divide PAM50 basal-like tumors into two groups**, and **combine HER2 and luminal B** while maintaining a **separate group of luminal B** PAM50 subtype tumors

The CoTC groups, composed of PAM50-classified basal-like tumors (CoTC1 and 2), are distinguished by differential expression of immune markers, E2F and MYC targets, along with G2M checkpoint-related proteins

### Proteome characterization reveals tumor subclass processes.

Protein abundances of a number of known complex members are exceedingly correlated

Although the common biological process involvements of protein complex members is expected to be evidenced by correlation21, the phy- sical nature of component interactions or their tightly related functions may exacerbate this effect

proteins with known interactions have more correlated abundances

these **elevated correlations are substantially more distinguished at the protein as opposed to the transcript level**

Correlation also appears to be **indicative of co-function**

proteins functioning as components of similar biological processes are highly connected 

Con- sidering each CoTC and PAM50 group individually and overlaying protein abundances onto this network provides snapshots of the defining characteristics of each group



### MET and EGFR are coregulated in normal-like tumors



we examined known **drug targets for correlating expression**, which would impli- cate them as operating in series/parallel and suggest that they are promising co-targets.

 highly correlated protein abundances of drug targets ->  suggesting they may operate in concert and render tumors sensitive to simultaneous targeting 

protein abundances of MET and EGFR are highly correlated -> their co-expression may be a marker for basal-like and normal-like tumors

**co-elevation of EGFR and MET appears to be confined to ductal carcinoma in situ (DCIS) regions for a subset of normal- like tumors**

EGFR-MET co-localization may confer an advantage for their in situ survival

**normal-like subtype is often overlooked** as a BC classification because of the tumors’ **close semblance to normal tissue and less aggressive luminal** tumors. Identifying histopatho- logical markers is an important step to ensuring patients are properly stratified into treatment regimens while revealing the inverse coregulation of EGFR and MET in invasive and DCIS regions provides fodder for therapeutic development within this understudied disease class



### RNA–protein correlation analysis. 



transcript quantity is widely used as a surrogate for protein abundance



Positive and significant correlations exist across tumors between 70% of the proteins quantified in the Oslo2 Landscape
cohort and their mRNA transcripts

and do not appear dependent on protein/mRNA half-life, average protein precursor area, or number of peptide spectral matches

but proteins known to rapidly **accumulate ubiquitin** groups upon inhibition of the proteasome27 have quantities **significantly less correlated with their transcript abundances** , suggesting the influence of **transcript regulation is buffered for those whose abundances are controlled at the protein level**



**structural ribosomal proteins** and those of the inner**-mitochondrial membrane** embedded electron transport chain (oxidative phosphorylation) are not as highly correlated with transcript quantity as are groups of **soluble metabolic and signaling/proliferation-related proteins**



protein products of transcripts profiled as part of **BC prognostic panels are significantly enriched for high mRNA to protein correlation**

-> suggest that the **robustness of clinical mRNA markers is due, at least in part, to being reliable protein surrogates** and demonstrates a clear link to protein phenotype

mRNA–protein correlation appears to have a non-linear dependence on variability 



Genes causally associated with cancer (COSMIC)28 and breast cancer29 display varied mRNA–protein correlations

-> some proteins should not be studied by mRNA expression serving as a surrogate measurement



**Highly proliferative** tumors (basal-like, HER2, luminal B, and/or high MKI67) have a tendency to **have more correlated proteomes and transcriptomes** than lowly proliferative tumors (luminal A, normal-like)

- high abundances of proteins involved in transcrip- tion, splicing, translation, and cell cycle being associated with high-tumor mRNA–protein correlation

- high abundances of the extracellular matrix and plasma proteins are linked to a poor tumor mRNA–protein correlation 

  

### Impact of CNAs is dampened at the protein level.

gene copy-number alterations (CNAs) are known to be associated with expression of their corresponding transcripts (i.e., associated in cis) in breast tumors

mRNA expression in cis is associated with CNAs, and the same effect, though dampened, appears to be present at the protein level 

Considering gain and loss effects, **a gene selected whose protein abundance is associated with a cis CNA is twice as likely to have both mRNA expression and protein abundance associated with that CNA** than a gene selected based on an mRNA–CNA association

-> supporting the hypothesis that CNA effects are attenuated at the protein level



The attenuation ofCNA effects may be explained by regulatory mechanisms at the protein level. 

-> a gaussian mixture model of protein abundance and mRNA expression correlations to CNAs

highly attenuated genes = gene subset with high mRNA–CNA and low protein–CNA correlations -> these genes accumulate ubiquitin at an elevated rate.



CNAs:

* distributed over the genome 
* many **gain** effects (mRNA and protein) localized to **chromosome 17,** and cover genes of the **HER2 amplicon**
* **loss** effects influence estrogen signaling and are concentrated in **basal tumors**

6 genomic losses associated with increased abundances at both the mRNA and protein level -> possible implementation of a compensa- tory mechanism



Curtis et al. classifies breast tumors (**n = 2000**) based on **cis associations of CNAs and mRNA expression** of **619 genes**. 

Of those that we quantified at the protein level in the Oslo2 Landscape cohort, 83% overlap with the set of genes having significant copy number to transcript or to protein abundance associations as determined by **the linear regression method implemented by Curtis** 

-> consistency: a cohort sized for a tractable high-quality proteome quantification study contains sufficient statistical power to reproduce findings from much larger cohorts



## Proteogenomics identifies candidate immunotherapeutic tar- gets



Translation of tumorigenic genomic aberrations -> tumor-specific proteins, whose immunoreactivity renders them ideal candidate antigens for targeted immunotherapies

integrated proteogenomics analysis workflow11,12 (-> cf Zhu et al. 2018):

* MS spectra are searched against databases of known peptides, SNPs, mutations, and theoretical peptides from genomic regions believed to be noncoding derived from the 6 reading frame translation of the entire genome (restricted based on peptide isoelectric point)
* Spectra matching known peptides and multiple genomic regions are filtered out along with single amino acid variants (SAAVs) not meeting stringent verification criteria imposed by SpectrumAI12 



* hundreds of peptides are
  identified mapping to genomic regions thought to be noncoding or intronic (pseudogenes, noncoding RNA), or corresponding to un-annotated alternative gene translations (exon extensions, 3′ and 5′ UTRs, exonic alternate open-reading frames) 
* two-thirds of which are identified by at least two PSMs
* these peptides have a similar pI distribution as known peptides
* 10% of the corresponding coding
  genomic loci are supported by mappings of at least two peptides
* RNAseq measurements and proteomic MS-spectra5,33 verify that many of these peptides are transcribed and translated in independent breast tumor cohorts (
* CAGE34 and ribosomal profiling35 reveal they are transcribed and translated in other systems (normal and cancer) (Supplementary
* genetic loci corresponding to peptides previously annotated as pseudogenes and noncoding RNA are more highly conserved than random loci with the same annotation
* 61 peptide identifications are verified by MS-spectra of synthetic peptides

30% (116) were predicted to bind MHC class I36 and not identified in MS data from normal tissue3

Patient-specific candidate immunotargets

Tumor specificity in the breast of proposed immuno- targets is evidenced by their absence in normal surrounding tissue as detected in a separate MS proteomics assay

These profiles are contrasted to those which suggest patient (rather than tumor) specificit



Identifying antigen targets in the proteome, as opposed to the transcriptome, may streamline development of immunotherapies, because the immune system is activated by protein fragments displayed on the MHC



These high-confidence identifications demonstrate the prominent role proteogenomic analyses of high- quality proteome spectra will have in the push towards individualized medicine.



## Single amino acids variants (SAAVs) impact protein abundances

certain **SAAVs of proteins**, naturally occurring or acquired via somatic mutations, are known **risk factors** for the development or prognosis/therapy response predictors of cancer

Thousands of these SAAVs match Oslo2 Landscape proteome spectra

have a similar ΔpI distribution as known peptides 

28 (out of 31 randomly selected) spectral identifications are confirmed with synthetic peptides (Supplementary

breast cancer driver genes are amongst proteins identified with SAAVs 



DISCUSSION

<u>insights:</u>

identifying a definitive **separation of basal- like tumors based on immune components**

postulating **reg- ulatory control of characteristic cancer hallmark genes by phos- phorylation** of specific groups of regulatory proteins

revealing the **glycolytic preference of proliferative tumor subtypes**, 

determining **copy-number alterations may be attenuated by protein- level regulatory mechanisms**, such as ubiquitinylation and sub- sequent degradation in the proteasome.

validating the often ambiguous mRNA-based **normal-like subtype**, both with an unbiased approach and the identification of **MET-EGFR coexpression as a biomarker**; potentially for the DCIS component

<u>with their proteogenomic pipeline:</u>

identify proteins in individual tumors, with high confidence, corresponding to un-described gene variants, noncoding regions, or regions thought to be noncoding due to poor genome anno- tation

postulate protein products of undescribed gene var- iants and noncoding regions are the consequences of cancer genome instability, and that they are strong **candidates of tumor- specific targets for immunotherapies**