Krug et al. 2020

Proteogenomic analyses of 122 primary breast cancers

The integration of mass spectrometry-based proteomics with next-generation DNA and RNA sequencing pro- files tumorsmore comprehensively. Here this ‘‘proteogenomics’’ approachwas applied to 122 treatment-naive primary breast cancers accrued to preserve post-translational modifications, including protein phosphoryla- tion and acetylation

Proteogenomics is an approach to tumor profiling that com-
bines next-generation DNA and RNA sequencing with mass spectrometry-based proteomics to provide deep, unbiased quantification of proteins and post-translational modifications such as phosphorylation (Ruggles



initial analysis: (Mertins et al., 2016).



largest cohort to date of BRCA samples that were acquired to minimize ischemic time, maximizing fidelity and reducing pre- analytical variability

the first comprehensive report of the BRCA acetylome



identifica- tion of 29,647 somatic mutations, 23,692 gene-level copy num- ber events, 23,121 gene transcripts, 10,107 proteins, 38,968 phosphorylation sites, and 9,869 acetylation sites



Non-negative Matrix Factorization-Based Multi-omics Classification of BRCA



single-omic and multi-omics clustering were performed for SCNA, mRNA, protein, and individual phos- phosite and acetylation site abundance using non-negative ma- trix factorization



To further probe NMF luminal cluster assignments, random
forest classifiers were trained on protein or mRNA data to distin- guish PAM50 LumA samples assigned to the NMF LumB-I clus- ter from PAM50 LumA samples assigned to the NMF LumA-I cluster. When



Integration of the cur- rent dataset with that of Johansson et al. (2019) supported NMF reassignment of some PAM50 LumA samples into the LumB-I group and suggested that their ‘‘basal immune’’ cluster was chiefly defined by an active immune microenvironment (Fig- ures S4B–S4E). 



phosphoproteomic data were used as kinase activation
surrogates



These pu- tatively activated kinases are candidates for subtype-specific treatment



Subtype-Specific Expression of Targetable, Highly Phosphorylated Kinases



Proteogenomic Metabolic Profiling and Acetylproteomics Highlight Subtype-Specific Metabolism



Ac levels normalized to protein abundance were used to identify NMF cluster-specific protein Ac events (



Unsupervised clustering of nuclear protein Ac revealed two subgroups of NMF-Basal-I tumors (Figure



Proteogenomics Analysis of ERBB2+ BRCAs



PG Analysis of the Immune Tumor Microenvironment (I- TME) Suggests Broader Applicability of Immunotherapy in BRCA



RNA-based immune cell deconvolution signatures and protein- level signatures for immune modulators (Thorsson et al., 2019) revealed a range of immune-related features across all four intrinsic subtypes (Table



acquired immune response hallmarks gener- ally appear not to be activated in LumA, whereas other PAM50 subtypes, including LumB, exhibit features consistent with acti- vation of acquired immunity



APOBEC-Mediated Mutagenesis Correlates with an Active I-TME in Luminal BRCA



APOBEC-driven mutagenesis can contribute to an active immune microenvironment in ER+ BRCA with links to PD-L1 mRNA expression.



Loss of SSBR Proteins Promotes Immunity in Luminal BRCA Gene



PG Analysis of Rb Status May Inform the Response to CDK4/6 Inhibitor Therapy



analysis of Rb provides a good example of how PG data integration could enhance prediction of drug efficac

knowledge of genomic Rb status could prove useful for repurposing CDK4/6 inhibitors for TNBC. However, TNBC samples often showed loss of Rb protein without a detect- able genomic aberration in the RB1 gene (Figure 6A). This raised the question of whether Rb protein estimates could contribute to prediction of CDK4/6 inhibitor activity when the RB1 status is wild type according to genomic analysis

inferred mTOR kinase activity was also higher in TNBC tumors with elevated Rb phosphoprotein levels compared with Rb-low tumors, suggesting activation of the PIK3-AKT-mTOR pathwa



Parsons et al. 2020



LumA and LumB breast cancers are both estrogen receptor (ER)-positive

ER target genes like cyclin-dependent kinase (CDK) 1 or the kinase Src promote cell proliferation, invasion and epithelial–mesenchymal transition (EMT)



LumB cancers have high expression of the proliferation marker Ki67, which correlates with increased risk of developing distant metastases (Colzani et al., 2014), and reduced expression of the progesterone receptor (PR) (Cho, 2016), which shifts gene expression toward more tumorigenic genes (Mohammed



LumB cancers have high expression of the proliferation marker Ki67, which correlates with increased risk of developing distant metastases (Colzani et al., 2014), and reduced expression of the progesterone receptor (PR) (Cho, 2016), which shifts gene expression toward more tumorigenic genes (Mohammed

treated using ER antagonists (e.g., tamoxifen), aromatase inhibitors and selective estrogen receptor degraders (e.g., fulvestrant). However, therapeutic resistance may arise through loss of ER expression, mutations in ER or overexpression of alternative breast cancer-driving pathways such as ERBB1/EGFR (Garcia-Becerra et al., 2012; Clarke et al., 2015; Ma et al., 2015). To overcome resistance to traditional ER antagonists targeted therapies against phosphoinositide 3-kinases (PI3K), mammalian target of rapamycin (mTOR), and CDK4/6 have recently been proven beneficial in the clinical setting (Beaver



HER2 + breast cancers overexpress ERBB2/HER2 (Iqbal
and Iqbal, 2014) which promotes proliferation by regulating CDKs and Cyclins (Timms et al., 2002). Additionally, HER2 dimerization with EGFR induces activation of mitogen-activated protein kinase (MAPK), c-Jun N-terminal kinases (JNK), and phosphoinositide phospholipase C (PLCγ) signaling pathways resulting in increased cell proliferation, migration and apoptosis resistance (Masuda et al., 2012).

treated with targeted agents such as trastuzumab, pertuzumab, and neratinib. Trastuzumab is an antibody which inhibits HER2 dimerization, promotes natural killer cell recruitment to tumors and stimulates ubiquitin-dependent HER2 degradation (Vu and Claret, 2012; McCann and Hurvitz, 2018; Schmid et al., 2018; Vikas et al., 2018). Therapeutic resistance to trastuzumab occurs via HER2 dimerization with other ERBB family members or constitutive HER2 activation



basal epithelial-like (BL) based BL breast cancers do not generally express ER, PR or
HER2 (Milioli et al., 2017), like triple negative breast cancers (TNBCs) (Lehmann et al., 2016) BLs are highly heterogeneous and include basal-like1-2, claudin-low, and immunomodulatory subgroups (Garrido-Castro et al., 2019). BLs have a highly proliferative and invasive phenotype with high risk of relapse in early breast cancer (Fallahpour et al., 2017) 

BLs are typically treated by chemotherapy and radiotherapy (Wahba and El- Hadaad, 2015) although recent advances have led to novel treatment opportunities for BL cancer patients. For instance, immunomodulatory BLs can be treated with immune checkpoint programed cell death protein 1 (PD-1) and poly (ADP-ribose) polymerase (PARP) inhibitors (McCann and Hurvitz, 2018; Schmid







Yu et al. 2021

integrative
analyses of multiple datasets with genomics, transcriptomics,
metabolomics, and single-cell transcriptome profiling.

Ten oncogenic pathways45 were selected to evaluate the enrichment score using ssGSEA.

scRNA-seq data processing The GEO: GSE75688 dataset contains annotated cell types from each sample. We determined diverse immune cell types in GEO: GSE114727 with the Seurat analysis package



The application offour normalization methods and the calculation of metabolic pathway activity were performed as previously described.31 The metabolic profiles were then compared between different cell types using GSEA.



Differential gene expression analysis was conducted using the limma package for METABRIC and the DESeq2 package for TCGA-BRCA in R software



Lytrivi et al. 2020



we combined RNA-sequencing of palmitate-treated human islets with iTRAQ proteomics of insulin-secreting INS-1E cells following a time course exposure to palmitate



This is the first study to combine transcriptomic and sensitive time course proteomic profiling of palmitate-exposed β-cells. Our



we combined the output of a random forests algo- rithm to infer regulations among the differentially expressed genes and proteins, and a prior network ob- tained



Rank-rank hypergeometric overlap (RRHO) Rank-rank hypergeometric overlap (RRHO) maps were generated by a modified two-tailed RRHO method [24,



Network inference analysis Gene regulatory networks were obtained by combining inferred networks from the expression profiles of the proteomics and RNA-seq datasets and a prior network, obtained from literature knowledge

A regulatory network was inferred in the RNA-seq and proteomic datasets separately.



A random forest algorithm was used to score predic-
tors of a target variable. A similar approach has been proposed in GENIE3 [87]. This was implemented in R using the package “randomForest” RF

Regulations (edges) were selected if p < 0.001 or alternatively if p < 0.05 and the regulation was present in the prior network. This

This analysis was performed for the 2 datasets (RNA-seq and proteomics) separately. The two obtained networks were then merged and a final network of 416 regulations involving 190 genes/ proteins was obtained.





Pozzoli et al. 2020

Once again, we create a graph, but in this case, it is used to encode
the participation of proteins in the protein complexes. Like before, spectral clustering is used to divide nodes, that is features, into candi- dates. We create a bipartite graph B where the vertices are 9995 proteins
on one side and 2916 protein complexes on the other side. A protein is connected to a protein complex if it is part of that protein complex. Since B is bipartite, the only edges in the graph are between proteins and protein complexes. Candidates have no protein complexes; therefore we need to encode
the further information added by the protein complexes in the weighted edges in the so-called feature graph. We generate a new graph, i.e., ′F , which is the weighted projection of B onto the proteins. There are 9995 nodes in ′F , one per protein. Two proteins u and v in ′F are connected with an edge if they share at least one common protein complex in B. Weight is directly proportional to the number of shared protein com- plexes in B, but in this particular case it is important to penalize those protein complexes common to many proteins because the signal of high-degree protein complexes is not as strong as the signal of low- degree protein complexes. So, we use Newman's weighted projection of B onto the proteins [18]. Weight



Li et al. 2015

We developed a method for gene association network construction based on gene expression data that integrate a variety of biological resources. 

Assuming gene expression data are from a multivariate Gaussian distribution, a graphical lasso (glasso) algorithm is able to estimate the sparse inverse covariance matrix by a lasso (L1) penalty. 

The inverse covariance matrix can be seen as direct correlation between gene pairs in the gene association network.

 In our work, instead of using a single penalty, different penalty values were applied for gene pairs based on apriori knowledge as to whether the two genes should be connected. The apriori information can be calculated or retrieved from other biological data, e.g., Gene Ontology similarity, protein-protein interaction, gene regulatory network.



even when some prior knowledge is not correct, the overall quality of the wglasso network was still greater than when not incorporating that information, e.g.,glasso.



gene expression is assumed to follow a mul- tivariate Gaussian distribution N(m, S), and Gaussian Markov ran- dom fields have been used to infer the structure of networks from
gene expression data



The Hammersley–Clifford theorem implies that zeros in the inverse covariance matrix of a multivariate Gaussian distribution indicate absent edges in the corresponding graphical model (Besag



Therefore, the problem of estimating the gene association network based on gene expression data can be transferred to estimating $\sum^{-1}$ or selecting nonzero entries in $\sum^{-1}$ 



Many studies, however, have used Pearson correlation coefficients between pairs of genes to infer network structure; Pearson correlation coefficients correspond to the covariance matrix $\sum$ and cannot infer the true structure



An accurate inference of biological network using Gaussian
graphical model is challenging for two main reasons. 

1. The first is that most genome-scale datasets are highly dimensional. Given p genes, there are possible p(p 2 1)/2 edges, but gene expression data often have a limited number of samples. When

2. The second major difficulty is the lack of efficient methods to
   integrate multiple levels of biological data to enhance model accuracy

The Lasso regression can be interpreted as a Bayesian regression
with a Laplace prior distribution, Lp(0, r)

In Bayesian statistics, prior information can be integrated into the model by changing the param- eter in the prior distribution, and the posterior distribution should better approximate the true distribution of the data. Therefore,



for the glasso algorithm, instead of using a single penalty parameter, it is reasonable to specify different amounts of penalties for different ele- ments in Q based on apriori information as to whether two genes are associated, or not. A smaller penalty can be given if apriori informa- tion indicates that they are linked



The prior matrix can be obtained in numerous ways. For example,
the GO semantic similarity between genes can be calculated using tools GOSemSim (Yu et al. 2010) or GOssTo (Yu et al. 2010), Then, the inverse of similarities can be implemented in the prior matrix, as a high similarity means a low penalty in our model. Other



Struyf et al.

s an interactive online resource with navigable
proteomics, transcriptomics, and drug sensitivity profiles at
https://lehtio-lab.se/forall/.

Our dataset demonstrates that proteomics and transcriptomics identify different phenotypic landscapes in childhood ALL cell lines, and we used proteomics to identify biological and phenotypic characteristics that suggest promising novel insights into both on-target and potential off-target drug activities.

Through proteomics-guided analysis of childhood ALL cell lines, we have generated a comprehensive resource of biomarkers and drug sensitivities, and identified a potential new therapeutic vulnerability to target the MEF2D- HNRNPUL1 rearranged high-risk subgroup





Herbst et al. 2020

unsupervised multi-omics factor analysis (MOFA) across all omics data sets in paralle



Schmidt et al. 2018

time-series RNA sequencing (RNA-Seq) and proteomics profiling on the same samples during human iTreg differentiation. To

Integrative analysis of the transcriptome and proteome confirmed involvement of specific molecular processes, as well as overlap of a novel iTreg subnetwork with known Treg regulators and autoimmunity-associated genes. 

We explored the transcriptome and proteome of the very same samples, enabling true integrative analysis of both data types and making deep quantitative mass spectrometry-based proteomics data of iTregs available to the scientific community. In

we provide a completely independent additional RNA-Seq dataset for TGF-β + ATRA iTregs generated in another laboratory under different culture conditions and with an extended time-series. 

We compared RNA and protein expression by matching RNA to the reference proteome, using the Ensembl gene ID as a key. We averaged log2Rs if multiple proteins mapped to the same gene. 

We created SOMs as implemented in the R package kohonen [92]. We started from a matrix of gene z-scores obtained from the RNA-Seq rlog-transformed data and selected all the DEGs according to the score G above. A supersom model was trained, considering seven different layers corresponding to the groups G01 to G07 above and a 20 × 20 toroidal hexagonal grid.



we performed unsupervised clustering using a version of Self-Organizing Maps (SOMs) that allows obtaining sev- eral parallel maps and perform data fusion (see Methods). As



Argelaguet et al. 2018

Multi-Omics Factor Analysis (MOFA), a computational method for discovering the principal sources of vari- ation in multi-omics data sets. MOFA infers a set of (hidden) factors that capture biological and technical sources of variability. It disen- tangles axes of heterogeneity that are shared across multiple modalities and those specific to individual data modalities. T

learnt factors enable a variety of downstream analyses, including identification of sample subgroups, data imputation and the detec- tion of outlier samples.



integration of OMICs data

1. A basic strategy for the integration of omics data is testing for
   marginal associations between different data modalities. A prominent example is molecular quantitative trait locus mapping, where large numbers of association tests are performed between individual genetic variants and gene expression levels (GTEx
2. A second strategy is the use of kernel- or graph-based methods to combine different data types into a common similarity network between samples



Given several data matrices with measurements of multiple omics data types on the same or on partially overlapping sets of samples, MOFA infers an interpretable low-dimensional data representation in terms of (hidden) factors (Fig 1A). 



These learnt factors capture major sources of variation across data modalities, thus facilitating the identification of contin- uous molecular gradients or discrete subgroups of samples

inferred factor loadings can be sparse, thereby facilitating the link- age between the factors and the most relevant molecular features.

MOFA disentangles to what extent each factor is unique to a single data modality or is manifested in multiple modali- ties (Fig 1B), thereby revealing shared axes of variation between the different omics layers. 



MOFA builds upon the statistical framework of
group Factor Analysis

1. fast inference based on a varia- tional approximation,
2.  inference of sparse solutions facilitating interpretation,
3. efficient handling of missing values and 
4. flex- ible combination of different likelihood models for each data modality, which enables integrating diverse data types such as binary-, count- and continuous-valued data. 



MOFA identifies important clinical markers in CLL and reveals an underappreciated axis of variation attributed to oxidative stress



MOFA identifies outlier samples and accurately imputes missing values



Latent factors inferred by MOFA are predictive of clinical outcomes



In an application to single cell data MOFA reveals coordinated changes between the transcriptome and the epigenome along a differentiation trajectory



https://github.com/bioFAM/MOFA. 

Code to repro- duce all the analyses presented is available at https://github.com/ bioFAM/MOFA_analysis.





Huang et al. 2017

For mRNA and protein expressions, expression values were collapsed across transcripts or isoforms to the corre- sponding gene using the highest mean when there were two transcripts or isoforms, or the value with the highest connectivity when there were more than three transcripts or isoforms as implemented in the WGCNA R package51



Mertins et al. 2016

e quantitative mass-spectrometry-based proteomic and phosphoproteomic analyses of 105 genomically annotated breast cancers, of which 77 provided high-quality data





Ma et al. 2019

Multi-view Factorization AutoEncoder (MAE) with network constraints that can seamlessly integrate multi-omics data and domain knowledge such as molecular interaction networks. 

Our method learns feature and patient embeddings simultaneously with deep representation learning. 

Both feature representations and patient representations are subject to certain constraints specified as regularization terms in the training objective. 

By incorporating domain knowledge into the training objective, we implicitly introduced a good inductive bias into the machine learning model, which helps improve model generalizability. 



To alleviate the overfitting problem in deep learning on multi-omics data with the “big p, small n” problem, it is helpful to incorporate biological domain knowledge into the model as inductive biases.



Besides net- work regularization approaches, directly encoding biolog- ical networks into the model architecture is also possible [20, 21], which usually requires subcellular hierarchi- cal molecular networks as the prior knowledge. Because high-quality human data is lacking (human biological interaction networks such as protein-protein interaction networks are still incomplete and noisy), network reg- ularization approaches are often preferable to directly encoding the noisy interaction network into the model architecture.



Since our model can learn a feature representation Y,
this representation should ideally be “consistent” with the biological interaction network corresponding to these fea- tures. We use a graph Laplacian regularizer to minimize the inconsistency between the learned feature representa- tion Y and the feature interaction network G:



minimizing the loss function can effectively reduce the inconsistency between the learned feature rep- resentation and the biological interaction network. The objective function incorporating biological interac-
tion networks through the graph Laplacian regularizer

the trace as surro-gate for measuring the inconsistency between the learned feature representation Y and the known feature interac-tion network G



we use a separate autoencoder for each view



We can get a patient similarity network S(v) for each view 

the outputs of multiple encoders can be “fused” together for supervised learning

Another approach is to concatenate all views together like DenseNet [30]. We have tried using both in our experiments and the results are not significantly different

can fuse them together using affinity network



Similar to the feature interaction network regularizer
(Eq. 6), we also include a regularization term on the patient view similarity:



For each type of -omic data, there is one correspond-
ing feature interaction network G(v). 

Different molecular interaction networks involve distinct feature sets and thus cannot be directly merged. 

However, patient similarity networks are about the same set of patients, and therefore can be combined to get a fused patient similarity network S using techniques such as affinity network fusion [14]. 

Our framework uses both molecular interaction networks and patient similarity networks for regularized learning.

We implemented the model using PyTorch (https:// github.com/BeautyOfWeb/Multiview-AutoEncoder



Ma and Zhang 2017

Similar to SNF, ANF treats each omic data type as one view of patients and learns a fused affinity (transition) matrix for clustering. 

We applied ANF to a carefully processed harmonized cancer dataset downloaded from GDC data portals consisting of 2193 patients, and generated promising results on clustering patients into correct disease types. 

Our experimental results also demonstrated the power of feature selection and transformation combined with using ANF in patient clustering. 

Moreover, eigengap analysis suggests that the learned affinity matrices of four cancer types using our proposed framework may have successfully captured patient group structure and can be used for discovering unknown cancer subtypes.



Zhang et al. 2020

Recently, graph neural networks (GNNs) have been successfully applied to graph-structured data such as point cloud and molecular data. These networks often only consider pairwise dependencies, as they operate on a graph structure. We generalize the GNN into a factor graph neural network (FGNN) providing a simple way to incorporate dependencies among multiple variables. We show that FGNN is able to represent Max-Product belief propagation, an approximate inference method on probabilistic graphical models, providing a theoretical understanding on the capabilities of FGNN and related GNNs. Experiments on synthetic and real datasets demonstrate the potential of the proposed architecture.



Liu et al. 2021

A proteomic and phosphoproteomic landscape of KRAS mutant cancers identifies combination therapies

a proteomic and phosphoproteomic landscape of 43 KRAS mutant cancer cell lines across different tissue origins

By integrating transcriptomics, proteomics, and phosphoproteomics, we identify three subsets with distinct biological, clinical, and thera- peutic characteristics. 

The integrative analysis of phosphoproteome and drug sensitivity information facili- tates the identification of a set of drug combinations with therapeutic potentials. 

we carried out a mass spectrometry (MS)-based quantitative proteomic and phosphoproteomic analysis of 43 KRAS mutant human cancer cell lines. By integrating multi-plat- form data types including proteome, phosphoproteome, and public transcriptome, we identified three subsets of KRAS mutant cancers. Molecular characterization of these three sub- sets enabled us to identify the prognostic difference and thera- peutic vulnerabilities of KRAS mutant cancers.

Proteomics and phosphoproteomics enable multi- omics-based integrative subtyping of KRAS mutant cancer cell lines 

To obtain a comprehensive understanding of the molecular heterogeneity of KRAS mutant cancers, we integrated our prote- omic and phosphoproteomic data with the publicly available transcriptomic data from the Genomics of Drug Sensitivity in Cancer (GDSC) database (Garnett

A fused similarity network of the three types of omics data was acquired using a similarity network fusion (SNF) method (Liu et al., 2018; Wang et al., 2014) after data imputation (Table

Unsupervised clustering for each respective layer of omics data showed that the average silhouette width of the subsets from transcriptome, proteome, or phosphoproteome alone was smaller than that of the subsets from the integrative multi-omics data (Figures S1F and S1I), suggesting that the integration of three types of omics data could better classify KRAS mutant cancers

assessment of the contribution to the top 20% fea- tures of normalized mutual information (NFM), as previously described

mutation types and copy number variation (CNV) of KRAS, as annotated by CCLE (Barretina

Multivar- iate analysis (**permutational multivariate analysis of variance** [**PerMANOVA**]) showed that the frequency of driver gene muta- tions (summed for each sample) was significantly correlated with subset (p = 0.026) rather than tissue origin

To identify the most discriminative molecular signatures
among these subsets, we extracted the most important mo- lecular features (i.e., variables) to discriminate the subtypes using a random forest learning method (Genuer

To further investigate the discriminative power of the identified
signatures in primary samples, we performed unsupervised clus- tering of a larger cohort (n = 507) of KRASmutant clinical samples from TCGA dataset using the signature mRNAs

Multivariate analysis (PerMANOVA) showed that the fre- quency of driver gene mutations (summed in each sample) was significantly correlated with both subset (p = 0.001) and tissue origin (p = 0.001) (Table

To explore the potential utility of the subtyping signatures, we
compared prognosis among the three clinical subsets. Disease-

unsupervised clustering of KRAS WTTCGA clinical
samples (including LUAD, LUSC, PAAD, COAD, and READ; n = 1,093) was conducted to assess the impact of tissue types.

Together, we showed that the identified subset signatures could be used to stratify the large and independent datasets of clinical samples

Bioinformatics reveals distinct molecular features of KRAS mutant subsets

To obtain mechanistic insights into different clinical outcomes of the three subset-defined tumors, we carried out gene set enrich- ment analysis (GSEA) using hallmark gene sets.

We further investigated the detailed pathways in each subset.

Overall, the results were consistent with those of the transcriptome and prote- ome data

To identify functionally important subset-specific phosphor-
ylation, we retrieved all the functionally annotated phospho- sites from the PhosphoSitePlus database

**Kinase-substrate enrich- ment analysis** 

Immunoblotting further confirmed the enrichment of EGFR signaling in S2 (Figure

illustrated the proteomic and phosphoproteomic alterations of frequently altered genes in a simplified RTK/RAS pathway diagram

Correlation between molecular feature and drug sensitivity reveals subset-specific therapies for KRAS mutant cancers

To explore the potential vulnerability of KRASmutant cancers, we compared gene dependencies among the three KRAS mutant subsets on the basis of theCancer Depen- dency Map (DepMap) dataset 

RNAi-based analysis identi- fied30genes with statistically different de- pendency scores  (**gene dependency analysis**)

To further identify the subset-specific therapies and their molecular indicators,
wecalculated thedrug-proteinand drug-phosphosite correlations in each subset.

. We further examined the subset-specif- ically upregulated proteins/phosphorylations of U.S. Food and Drug Administration (FDA)-approved drug targets from the Drug- Bank database (unpaired two-sided Student’s t test) (Figure S4J), suggesting a therapeutic potential worthy of further investigation.

Phosphoproteome analysis reveals drug combinations for the treatment of KRAS mutant cancers

KRAS mutations activate multiple downstream pathways. Previ- ous studies have shown that blockage of individual downstream pathways barely achieves therapeutic benefits (Manchado et al., 2016), suggesting a possibility of targeting signaling compensa- tion for KRAS mutant cancer treatment

We therefore raised a hypothesis that two drugs showing a large number of inverse correlations to the same set of phosphoproteins (i.e., protein phosphorylation levels) might have a higher chance to show synergistic effects, in particular when these phosphoproteins are co-activated in KRAS mutant cancer

we conducted a **compensatory phosphoprotein biomarker analysis** (CPBA) (Fig- ures 5A and S5A). In this analysis, we introduced a combination score to evaluate the compensatory effects, which was calcu- lated by multiplying the number of positively correlated phos- phoproteins with the number of negatively correlated phospho- proteins in each drug combination

CPBA may be a useful approach to predict the ef- ficacy of drug combinations for KRAS mutant cancers

Combined inhibition of DOT1L and SHP2 shows specific therapeutic benefit for S2 tumors

Among the new drug combinations, we noted a pair that involves an inhibitor DOT1L plus an inhibitor of SHP2 (DOT1Li-SHP2i combo). This combination was particularly appealing, as the **crosstalk between epigenetic regulation and phosphoproteome** has been noted in cancer treatment (

To explore the predictive power of theS2 signatures, we tested the combination in 41KRAS mutant cell lines evenly distributed in three subsets

These results suggested thepredictive utility of our signature to stratify KRAS mutant cancer to DOT1Li- SHP2i combo therapy independent of tissue origins

As CPBA was based on the upregulated phospho-signaling in
KRAS mutant cells versus the WT ones, this approach was ex- pected to only predict the sensitivity to KRAS mutant tumors

, the therapeutic efficacy of DOT1Li-SHP2i combo
was assessed in vivo. I

Similar results were observed in a larger panel of cell lines retrieved from GDSC database

DOT1L inhibition results in an induced vulnerability to SHP2 inhibitors in KRAS mutant cancers We

We next examined the molecular basis underlying the therapeu- tic benefit of DOT1Li-SHP2i combo.

Immunoblotting analysis showed that both multiple DOT1L inhibitors and the genetic depletion of DOT1L resulted in the upregulated MAPK and PI3K/AKT

epigenetic modulations could
give rise to a targetable vulnerability

DOT1L inhibition upregulates MAPK/AKT pathways via transcriptionally regulating PREX1 

To investigate how DOT1L inhibition causes feedback upregula- tion of the MAPK/AKT pathways, we performed a proteomic analysis to systematically examine the proteins affected by DOT1L inhibition
