### Single-cell analyses reveal key immune cell subsets associated with response to PD-L1 blockade in triple-negative breast cancer - Zhang et al. 2021



combine **single-cell RNA- seq, TCR-seq, and ATAC-seq** to investigate **immune cell dynamics** in the **tumor microenvironment and peripheral blood** of patients with TNBC treated with **paclitaxel or paclitaxel plus atezolizumab**, revealing immune features of responders and nonresponders, the mechanisms and intertwined effects of paclitaxel and atezolizumab in TNBC treatment.

* Dynamic **immune cell alterations** revealed in TNBC following anti-PD-L1 treatment
* **CD8-CXCL13 and CD4-CXCL13 T cells predict effective responses** to PD-L1 blockade
* Proinflammatory macrophages expressing CXCL9 are associated with CXCL13+ T cells
* **Paclitaxel impairs the expansion of responsive immune cells caused by atezolizumab**

single-cell RNA- and ATAC-sequencing to examine the immune cell dynamics in 22 patients with advanced TNBC treated with **paclitaxel or its combination** with the **anti-PD-L1 atezolizumab**

high levels of baseline **CXCL13+ T cells** are linked to the proinflammatory features of macrophages and can **predict effective responses to the combination therapy.**

In **responsive** patients, lymphoid tissue inducer (LTi) cells, follicular B (Bfoc) cells, CXCL13+ T cells, and conventional type 1 dendritic cells (cDC1) concertedly **increase following the combination therapy, but instead decrease after paclitaxel monotherapy**.

the i**mportance of CXCL13+ T cells in effective responses to anti-PD-L1 ther- apies** and suggest that their reduction by paclitaxel regimen may compromise the clinical outcomes of accompanying atezolizumab for TNBC treatment.

**Immune checkpoint blockade** (ICB) targeting immunoregulatory pathways such as PD-1/PD-L1 signaling axis has

the fundamental mechanisms un- derlying ICB resistance remain poorly understood

**T cells** in TME often reach a **dysfunctional or exhausted state**, which is characterized by the **overexpression of immune checkpoint molecules** and the reduction of effector function

innate and adaptive immune cells in TME act in concert to control tumor growth

modulation of other immune cells for T cell responses

ICB shows promise in TNBC treatment, clinical studies have observed a modest response

exploration of ICB in combination with chemotherapy.

**combining nab-paclitaxel with atezolizumab, an anti-PD-L1 antibody, can prolong progression-free survival** of patients with metastatic TNBC,

approved for the treatment of patients with metastatic or locally advanced unresectable TNBC

however the combination of paclitaxel with atezolizumab fails to meet the primary endpoint in the IMPASSION131 trial (Miles

incom- plete understanding of the underlying mechanisms of such a combination therapy

, we leverage **single-cell RNA-sequencing (RNA-seq) and ATAC-sequencing (ATAC-seq)** technologies to characterize the cellular and molecular dynamics of immune cells in **22 patients with TNBC treated with paclitaxel or its combination with atezolizumab.**

* uncover the underpinnings of resistance and sensitivity to anti-PD-L1 therapies, 
* identify responsive immune cell subtypes to atezolizumab, 
* suggest the potential draw- backs of paclitaxel in combination with atezolizumab in TNBC treatment

<u>Immune cell dynamics in patients with advanced TNBC</u>

half of whom were treated with paclitaxel monotherapy and the other half with paclitaxel plus atezolizumab as first-line therapy

**responders showed higher levels of TILs and PD-L1, not all tu- mors with such features responded to anti-PD-L1 therapy**

single-cell sequencing to characterize immune features and dy- namics in patients with TNBC showing distinct clinical out- comes.

78 tumor biopsies and blood samples were collected at baseline, 4 weeks after treatment initiation, and dis- ease progression (Figure

exome sequencing; concordant with those commonly identified in hu- man breast tumors from The Cancer Genome Atlas (TCGA) (Ko-

single-cell transcriptome data for **489,490 high-quality im- mune cells** that included myeloid cells, innate lymphoid cells (ILCs), T cells, and B cells

**B cells were mainly enriched in responsive** tumors, while m**yeloid cells were enriched in nonresponsive** tumors tumors of the combination therapy group (Figure

**, B cells decreased while T cells increased after effective treatment** in both groups. We

two indices, ), to systematically investigate the connec- tion of distinct immune cell types with clinical responses

(1) the predictive index (Pi) = correlation between baseline cellular propor- tions with tumor size changes,

* positive Pi =  a higher level of baseline in cellular proportion of the corresponding immune cell subtype associated with a higher level of decrease in tumor size following treatment -> predicting favorable response
* negative Pi = the corresponding immune cell subtype predicts an unfavorable response

**B cells** were the most prominent immune cell type that could **predict a favorable response** to both treatment regimens; predictive roles of B cells for ICB response

(2) the therapeutic index (Ti) = corre- lations between cellular proportion dynamics with tumor size changes 

* positive Ti = increase in cellular proportion of the corresponding immune cell subtype associated with a higher level of decrease in tumor size following treatment -> mediating a favorable response
* negative Ti = the corresponding immune cell subtype mediates an unfavorable response

**T cells** notably exhibited a **positive Ti for the combination therapy** in contrast to chemotherapy, suggesting that **atezoli- zumab may contribute to the effective response through elevating T cells.**

reclustered each major immune compartment derived from blood or tumors, separately; **heterogeneous immune cell subtypes** that exhibited **distinct molecular signatures**, indicative of their **unique cellular identities**

<u>CD8-CXCL13 and CD4-CXCL13 expanded in TNBC tumors responsive to the combination therapy</u>

2 T cell clusters (CD8-CXCL13, CD4-CXCL13) that highly expressed PDCD1 and CXCL13

* their **exhaustion features implied their tumor-reactivity** (**tumor-reactive T cells exhibit exhaustion phenotypes due to persistent tumor antigen stimulation**)
* accompanied with T regulatory cells (Tregs), showed remarkable **clonal expansion** based on matched T cell receptor (TCR) data, further indicating their **antigen-experienced properties** 

3 T cell clusters including naive (Tn), activated (Tact), and proliferative (Tprf) subsets that contained both CD8+ and CD4+ T cells

several canonical CD8+ or CD4+ T cell clusters including regulatory (Treg), central memory (Tcm), effector (Teff), effector memory (Tem), tissue-resident memory (Trm), and mucosal-associated invariant T (MAIT) cell subsets

decipher immune cell dynamics following treatment -> investigate the modulation of paclitaxel on distinct T cell subsets

* following paclitaxel treatment, in responders: 
  * decreased levels of CD8-CXCL13, CD4-CXCL13, and Tregs
  * elevated levels of CD8+ Tem, Teff, and CD4+ Tcm cells  
  * reduction of CD8-CXCL13 and CD4- CXCL13 also demonstrated by Ti analysis

* following paclitaxel + atezolizumab treatment, in responders:
  * both CD8-CXCL13 and CD4-CXCL13 expanded; 
  * expansion also demonstrated by Ti analysis

Such observations implied the pertinence of CXCL13+ T cells with PD-L1 blockade

TCR sequences to trace the lineage origins of CXCL13+ T cells in tumors respon- sive to the combination therapy

* their expansion contained both preexisting and newly infiltrated clones
* number of preexisting clones > newly infiltrated 

-> considerable preexisting clones contrib- uted to the expansion of CXCL13+ T cells following treatment

**association of CD8-CXCL13 with ICB therapies** confirmed by other published datasets

<u>CXCL13+ T cells and associated B cells predicted favorable responses to anti-PD-L1 immunotherapy</u>

responders to combination therapy: higher levels of baseline CD8- CXCL13 and CD4-CXCL13 -> **importance  of CXCL13+ T cells in shaping clinical outcomes of anti-PD-L1 therapy**; also:

* Pi analysis: **higher levels of CXCL13+ T cells could predict better responses** to the combination therapy 

* In addition to CXCL13+ T cells, **Tn-LEF1 also linked to a favorable response** to both treatment regimens

* TCGA data: breast tumors with fewer infiltrating CD8+ CXCL13+ T cells than melanoma and lung tumors, in line with the lower response rates to ICB for BC
* IHC staining confirms the presence of CD8-CXCL13 and CD4-CXCL13 T cells in tumors responsive to the combination therapy

2 major subpopulations of B cells in TNBC tumors

1. CD19+ B 
   * combination therapy: higher levels of baseline in responders
   * high expression of CXCR5, the receptor of CXCL13 (BCA-1) -> potential connections with CXCL13+ T cells
     * responders to combination therapy: colocalization  CD19+ B cells with CD8-CXCL13 and CD4-CXCL13 T cells
     * responders to ICB treatment in BCC: CD19+ B cells co-expanded with CD8+ Tex cells 
     * TCGA BRCA: high expression of B cell makers of CD79A and CD19 in tumors harboring substantial CD8+ CXCL13+ T cells 
2. plasma B (pB) cells

transcriptomic differences of B cells responsive vs nonresponsive tumors

* responsive: high expression of genes involved in pathways of antigen processing and presentation, response to interferon (IFN)-g, and T cell activation
* nonresponsive:  high expression of genes involved in pathways of B cell activation, immunoglobulin production, and humoral immune response 

=> **the function of B cells in antitumor immunity may be mainly attributed to activating CXCL13+ T cells**.



<u>Bfoc and LTi cells concertedly expanded in TNBC tumors responsive to the combination therapy</u>

CD19+ B cells comprised heterogeneous subsets of 

* naive B (Bn)
* memory B (Bmem)
* Bfoc
* proliferating Bmem
* roliferating Bfoc 

that showed no particular tissue enrichment

responders to paclitaxel + atezolizumab vs not responders to paclitaxel: enrichment of Bn, Bmem, and Bfoc cells 

responders in both treatment groups:

* higher levels of baseline Bfoc 
* Pi analysis: connection of Bfoc with favorable clinical responses

B cell dynamics, Bfoc:

* paclitaxel responders: decreased 
* paclitaxel + atezolizumab responders: increased

-> the pertinence of Bfoc cells to atezolizumab treatment

prognostic significance of distinct B cell subsets: Bfoc as the most significant B cell subtype associated with favorable clinical outcomes (BRCA TCGA data)

=> **Bfoc as the most important B cell subset for patient survival and ICB-immunotherapy**

connection of Bfoc with other immune cells:

* high correlations in cellular proportions with CD4-CXCL13 and CD8-CXCL13, especially in the posttreatment group 

  -> **Bfoc may interact with CXCL13+ T cells** to play a role in antitumor immunity

  * CXCR5 showed the highest expression level in the Bfoc subset, and multiple pairs including CD40LG-CD40, interleukin (IL)21-IL21R, and IL6ST-EBI3 might mediate the crosstalk between Bfoc and CD4-CXCL13

* high correlation also with LTi-like ILC3 (ILC3-IL7R), featured by the expressions of CCR7, IL7R, GPR183, LTB (lymphotoxin), and IL23R

  * concertedly expanded with Bfoc after the combination therapy but not after chemo- therapy
  * ILC3-IL7R could predict a favorable response to both treatment regimens -> potential role in effective antitumor immunity

<u>cDC1 expanded in TNBC tumors responsive to the combination therapy</u>

importance of CXCL13+ T cells in anti-PD-L1 therapy -> investigate how they might be activated and en- riched in certain patients with TNBC

**myeloid cells in TME can modulate T cell responses** -> focus on the myeloid compart- ment in TNBC tumors -> observe DCs, macro- phages, monocytes, and mast cells

* DCs:
  * higher levels in responders both at baseline and after treat- ment with the combination therapy but not chemotherapy -> supporting their **roles in ICB therapy**
  * unsupervised clustering to identify DC subsets (based on signature genes)
  * higher levels of baseline cDC1 and pDC could suggest favorable responses to paclitaxel, but none of these DC subsets could indicate favorable responses to the combina- tion therapy -> DC subsets likely not key determi- nants of effective anti-PD-L1 therapy 
  * after paclitaxel, responders: 
    * decreased levels of mDC; a temporal pattern resembling those of potential tu-mor-reactive T cells in the same treatment group 
    * mDC tightly connected with cDC1, due to their tran- scriptional similarity and correlated dynamic changes in cellular proportions
  * after paclitaxel + atezolizumab, responders:
    * also decreased mDC 
    * increased cDC1 and pDC
    * likely due to selective reduction of mDC by paclitaxel regimen
  * paclitaxel vs. paclitaxel + atezolizumab, responders to combination therapy:
    * higher levels of cDC1, mDC, and pDC -> indicating involvement in anti-PD-L1 therapy 
    * Ti analysis: cDC1 and pDC associated with favorable response to combination therapy but not chemotherapy 
    * reduction of mDC following paclitaxel treatment subdued when atezolizumab was also used -> indicating its pertinence to PD-L1 blockade
  * both mDC and cDC1 exhibited coordinated temporal dynamics with CD4-CXCL13 
    * confirmed by TCGA bulk RNA-seq data 
    * multiple ligand-recep- tor pairs that might mediate the crosstalk among mDC, cDC1, and CD4-CXCL13
    * CD8-CXCL13 also exhibited high correlations with mDC, cDC1, and CD4-CXCL13 

=> **CD4-CXCL13 and CD8-CXCL13 might be both activated by cDC1 or mDC**, as hinted by recent studies showing that cognate CD4+ T cells license cDC1 to induce CD8+ T cell immunity 

<u>The phenotypes of macrophages were associated with T cell states</u>

monocyte/macrophage compartment:

* combination therapy: nonresponders 
  * higher baseline levels of macrophages (Mf) 
  * enriched macrophages highly expressed SPP1, TREM2, FN1, and C3,
    * TREM2 and SPP1 previously found expressed in anti-inflamma- tory tumor-associated macrophages (TAMs) 
  * macrophages enriched in responders highly expressed CXCL9, MMP9, and PLA2G2D

unsupervised clustering re- vealed extensive heterogeneity (multiple subsets of monocytes and macrophages)

understand how CXCL13+ T cells were modulated by other immune cells -> TCGA BRCA data to examine the transcriptional features of breast tumors showing high expressions of CD8A and CXCL13 -> identification of a group of immune-related genes, top-ranking:

* 2 macrophage subsets
  * Mf-CCL2
    *  high expression of CXCL9, CXCL10, and CXCL11 
  *  and Mf-MMP9
    *  high expression of PLA2G2D, IL2RG, and CXCL9
  *  CXCL9 and CXCL10 are che- mokines known to mediate T cell recruitment -> support the **proinflammatory properties of these 2 macrophage subsets**
* mDC and other DC subtypes

-> suggest **connections with CXCL13+ T cells**



also examined immune cell proportions: Mf- CCL2 and Mf-MMP9 highly correlated with CXCL13+ T cells

higher baseline levels of these two macrophage subsets suggested a favorable response to the combination therapy (Figures

key signature genes of Mf-CCL2 and Mf-MMP9 associated with IFNG and TNF -> proin- flammatory phenotypes of macrophages might be sculpted by active antitumor immune responses in TME

paclitaxel + atezolizumab responders: simultaneous enrichments of Mf-MMP9 and CD8-CXCL13 

other subsets:

* baseline levels of Mf-MKI67, Mf-SLC40A1, and Mf-MGP positively correlated with Tregs
* Mf-CX3CR1, Mf-TUBA1B, and Mf-FOLR2 negatively correlated with CXCL13+ T cells
* subsets resembling SPP1+ TAMs ->  anti-inflammatory macrophages
* Pi analysis: enrichments in combination therapy nonresponsders

dynamic change analysis:

* paclitaxel, responders:
  * selective **decrease of proinflammatory** macrophages of Mf-CCL2 and Mf-MMP9
  * **increase of immunosuppressive** macrophages in responsive patients, 
* atezolizumab: little effect on macrophage subsets 

=> **paclitaxel may confer an immunosuppres- sive TME**

key **signature genes of CXCL13+ T cells and Mf-MMP9** by con- structing **scoring classifiers**, and revealed **a set of markers showing high predictive potentials**

=> **key signature genes of core antitumor immune subsets could be explored as biomarkers to predict ICB responses**

higher levels of myeloid cells in blood:

* monocytes largely predominant
*  could pre- dict a favorable response to both treatment regimens

transcriptomic fea- tures of baseline blood-derived myeloid compartment in combination therapy:

* suppressive TAM-associated genes enriched in nonresponders 
* CD74 and HLA class II genes enriched in responders 

->  monocytes in blood could partially reflect the im- mune features in TME

dendritic cells in blood:

* cDC1 not detected
* increase of cDC2 following the combination therapy in responsders 



<u>CD8-CXCL13 showed enhanced cytotoxicity following anti-PD-L1 therapy</u>

examined their treatment-induced transcrip- tional dynamics of CXCL13+ T cells (importance in anti-PD-L1 therapy !)

* CD8-CXCL13 showed lineage connection with CD8+ Tem cells
* upregulation of 
  * effector and memory-related genes 
  * transcription factors (TFs) of TBX21, BHLHE40, and BCL6
  * co-stimulatory molecules of CD28 and ICOS
  * HLA genes and integrins
* downregulation of 
  * exhaustion-related genes
* **phenotypic shift of CD8-CXCL13** also reflected by decreased exhaustion score and increased effector memory score

* upregulated genes:
  * involved in pathways of positive regulation of T cell–mediated cytotoxicity, antigen processing and presenta- tion, and IFN-g–mediated signaling pathway

=> suggest **augmented effector property after PD-L1 blockade treatment**

chromatin accessibility of signature genes for different subsets confirmedtheir cellular identities 

scATAC-seq analysis:

* en- richments of ATAC-seq peaks in T cell exhaustion markers in CD8-CXCL13 cells

* Compared with other T cell subtypes, CD8-CXCL13 and CD8+ Tem profiles more similar to each other -> confirm connections in a continuous spectrum of lineage differentia- tion

paired scATAC-seq data before and after the combination therapy.

* **augmented effector and memory properties**; diminished accessibilities in exhaustion-related genes
* overrepresented TF binding motifs (TFBMs) in differentially accessible peaks also confirmed the phenotypic change of CD8-CXCL13

chromatin accessibility dynamics consistent with transcriptome-based analyses

=> support the **enhanced effector properties of CD8-CXCL13 following the combination therapy**

investigation of T cells in blood

* proliferating T cells were enriched in responders a baseline, but decreased following the combination therapy, possibly due to the impact of paclitaxel

* no CXCL13+ T cells in blood, but blood CD8+ Tem cells shared higher levels of TCR clones with tumor CD8-CXCL13, and also elevated migration index in responsive patients following the combination therapy

  ​	->  detectable systemic immune response was induced in responsive patients following PD-L1 blockade treatment

<u>Discussion</u>

Bfoc but not pB cells expanded concertedly with LTi cells, CXCL13+T cells, and cDC1 following paclitaxel plus atezolizumab treatment

**B cells might contribute to antitumor immu- nity by interacting with and activating CD4-CXCL13 that could help prime CD8-CXCL13 through cDC1**.

high levels of CXCL9 and CXCL10, chemokines linked to effective ICB re-sponses,  could indicate the presence of CXCL13+ T cells in TME

**paclitaxel regimen could selectively reduce key antitumor immune cells but elevate immunosup- pressivemacrophages** inTNBCtumors -> **might not benefit the accompanying atezolizumab**

reduction of key antitumor immune cells might be caused by the direct andselectivedamagebypaclitaxel, likely due to the pro- liferative states of potential tumor-reactive T cells 

steroid premedication necessary procedure for the paclitaxel regimen, and it has been reported that steroidpremedicationcouldweakenantitumor immuneresponses

The 2 different treatment regimens could induce **opposite dynamic changes** **of key antitumor immune subsets** **only in responsive patients who had high levels of baseline CXCL13+ T cells.**

**paclitaxel may compromise the efficacy of accompanying atezolizumab**

* intrinsic properties of TNBC tumors, especially baseline level of CD8-CXCL13 T cells, seem to be the key de- terminants of clinical responses and the dynamic patterns of im- mune cells following different treatments
* pa- tients with high levels of baseline CXCL13+ T cells: 
  * can tolerate the deleterious effect of paclitaxel regimen, 
  * thus become responders and further increase the levels of CXCL13+ T cells after the combination therapy

* patients with low levels of CXCL13+ T cells
  * the effect of atezolizumab would fail to overcome the deleterious effect of paclitaxel regimen

un- clear whether and how paclitaxel will affect other ICB antibodies,

breast tumors were collected in the chemotherapy group, while metastatic tumors were collected in the combination therapy group since these pa- tients hadalreadygonethroughbreast tumor surgical removal. Im- mune infiltration levels are in general different in distinctmetastatic sites, whichmay affect the clinical outcomes of immunotherapies





<u>Dimension reduction and unsupervised clustering for scRNA-seq data</u>

standard workflow in Scanpy:

* top 4,000 highly variable genes (HVGs) detected using the dispersion-based methods 
  * normalized dispersion obtained by scaling with the mean and standard deviation of the dispersions for genes falling into a given bin
* Unwanted sources of variation including total counts, percentages of mitochondrial gene counts and heat shock protein (HSP) related gene counts further regressed out from the normalized expression matrices
* PCA on the variable gene matrix to reduce noise -> **top 50 components used for down- stream analyses**
* **correct batch effects** from different patients: BBKNN to **generate a batch-balanced k nearest neighbor (KNN) graph**, which identifies the top neighbors of each cell in each batch separately instead of the entire cell pool 
* Leiden algorithm applied on nearest neighbor graphs to **detect communities and find cell clusters**
* the same principal components used for **non-linear dimension reduction** to generate UMAP or t-SNE projections for visualization

the **first-round of unsupervised clustering**

* annotate each cell cluster according to canonical immune cell markers
* identifiy major immune cell types including T cells, ILCs, B cells and myeloid cells

immune cell clusters derived from tumor tissue and peripheral blood separated clearly -> **second-round of unsupervised clustering** on each major immune cell compartment derived from tumor tissue or blood, respectively -> obtain the high-resolution map of immune cells 

* same as the first-round clustering

both clustering:

*  started from unfiltered expression matrix
* identified HVGs
* calculated PCA matrix
* corrected batch effects by BBKNN
* detected cell clusters by Leiden algorithm and
* performed dimensionality reduction for visualization

**Marker genes detection**  using the rank_genes_groups function

**DEGs detection** in specific immune cell clusters:

* normalized raw count matrix by sctransform method
  * models the UMI counts using a regularized negative binomial model to remove the variation of sequencing depth, while adjusting the variance based on pooling information across genes with similar abundances
* limma with the normalized expression obtained via sctransform

<u>Dimension reduction and unsupervised clustering for scATAC-seq data</u>

**latent semantic indexing** (LSI) practice applied on the high-quality peak count matrix for downstream analyses

* term frequency-inverse document frequency **(TF-IDF) normalization performed to correct differences in sequencing depths** across cells and across peaks
* top 25% peaks were selected with the FindTopFeatures function 
* dimensional reduction with singular value decomposition **(SVD) on the TF-IDF matrix**
  * 1st LSI component removed for its correlation with sequencing depth
  * the other **top 30 LSI used to construct a shared nearest neighbor (SNN) graph** by first determining the k-nearest neigh- bors of each cell with k=20, and then calculating the neighborhood overlap between each cell and its k-nearest neighbors 
* smart local moving **(SLM) algorithm applied for modularity optimization and identified cell clusters**
* same LSI components also used for UMAP and t-SNE 
* annotation of cell clusters 
  * generated a gene activity matrix by counting the fragments intersecting the gene body and promoter (defined as 2 kb upstream) regions based on the chromatin accessibility data 
  * gene activity matrix normalized by the total feature counts, scaled with the median feature counts, and log- transformed.
  * integrated scATAC-seq data with paired scRNA-seq data according to the integration procedure of Seurat 
    * cross-modality integration with canonical correlation analysis **(CCA) for dimension reduc- tion and label transfer** with projected LSI for anchor building, based on the gene activity matrix and the gene expression matrix. 
  * To obtain cell type annotations for scATAC-seq clusters
    * calculated the confusion matrix based on their predicted labels
    * annotated each scATAC-seq cluster as the cell type with the highest cell number
    * cellular identities of each cluster confirmed with their marker gene activities
    * signature peaks of each cluster detected and annotated with the closest genes 
  * differentially accessible regions between cell clusters detected by a **differential accessibility (DA) test**
    *  logistic regression with the total number of fragments as a latent variable to mitigate the effect of differential sequencing depths



<u>Definition of predicative index and therapeutic index</u>

To circumvent the limitation of sample sizes: Pi and Ti to leverage the single-cell and clinical data in a continuous axis -> detect consistent patterns among immune cell compositions and connections with treatment-induced clinical outcomes 

*Pi* 

* to **determine whether a baseline cellular component is associated with the clin- ical response**
* measures the **correlation between baseline cellular proportions and treatment-induced changes in tumor size**
  * baseline cellular proportions = fractions of immune cell clusters in corresponding major immune compartments that involve T cells, B cells, ILC and myeloid cells, in pre-treatment samples
  * tumor size changes = relative changes of tumor sizes between post- and pre-treatment samples

$Pi = \frac{-slope}{\abs{slope}} * R^2$

* $slope$ and $R^2$ obtained by a linear regression model $lm(y1\sim x1)$ that estimates the correlation between the baseline cellular pro- portions ($x1$) of corresponding immune cell clusters with the tumor size changes ($y1$) in patients treated with different regimens
  * positive Pi = higher baseline level of the corresponding immune cell cluster associated with better clinical response
  * negative Pi = higher baseline level of the corresponding immune cell cluster associated with a worse clinical response
* based on the **baseline levels** -> derived from patients with **pre-treatment biopsy** data.

*Ti*

* to **determine whether a dynamic change in cellular proportion associated with the clinical response**

* measures the **correlation between the treatment-induced cellular proportion changes and tumor size changes**. 
  * cellular propor- tion dynamics = relative changes in cellular proportions for immune cell clusters in corresponding major immune compartments between post- and pre-treatment samples
  * tumor size changes = relative changes in tumor sizes between post- and pre-treatment samples

$Ti = \frac{-slope}{\abs{slope}} * R^2$

* $slope$ and $R^2$ obtained by a linear regression model $lm(y2\sim x2)$ estimates the correlation between the cellular proportion changes ($x2$) of the corresponding immune cell cluster with the tumor size changes ($y2$) in patients treated with different regimens.
  * positive Ti = higher level of increase in cellular proportion for the corresponding immune cell cluster associated with a better clinical response
  * negative Ti = higher level of increase in cellular proportion for the corresponding immune cell cluster associated with a worse clinical response

* Ti based on the **relative dynamic changes** of immune cell compositions between post- and pre-treatment samples -> derived only from patients with **paired biopsy** data





The <u>exhaustion score</u> was defined as the average expression of CXCL13, HAVCR2, PDCD1, TIGIT, LAG3, CTLA4, LAYN, RBPJ, VCAM1, GZMB, TOX and MYO7A, while <u>effector memory score</u> as the mean expression of effector- or memory-related genes of PRF1, IFNG, CCL4, HLA-DQA1, GZMK, GZMA, GZMH, CD44, DUSP2, KLRB1, KLRD1 and CTSW



<u>Tissue enrichment and time-point distribution analysis</u> 

quantify the **enrichment of immune cell types across different tissues and time-points** of sample collection

* compared observed and expected cell numbers for each cluster in each tissue and time-point 

$R_{o/e} = \frac{Observed}{Expected}$

* expected cell numbers of immune cell clusters in a given tissue or time-point calculated from the Chi-square test. 
* one cluster enriched in a specific tissue or time-point if $R_{o/e} >1$

<u>Predictive performance evaluation for signature genes</u> 

**identify potential biomarkers applicable in clinical practice for anti-PD-L1 immunotherapy**

* focus on immune cell subsets showing high predictive potentials and their signature genes
* Pi analyses that suggested positive connections of CXCL13+ T cells and their highly correlated immune cell subsets with the favorable responses to the combination therapy
* enrichments of immune cell subsets in baseline responsive patients compared to non-responsive patients 
  * Chi-Square test, after filtering immune cell clusters that contained less than 50 cells in baseline responsive samples
  * ratio of observed to expected cell numbers ($R_{o/e}$) to estimate the enrichments of immune cell clusters in baseline responsive patients
  * selected those with $R_{o/e}\ge1.2$ and false discovery rate (FDR) < 0.01 as immune cell clusters showing significant enrichments in responders
  * such enrichment analysis in both chemotherapy and the combination therapy groups
* **Immune cell clusters** showing **high Pi and significant enrichments** in responders defined as those with **high predictive potentials**
  * CXCL13+ T cells and Mf-MMP9 also showed consistent patterns in their connections with favorable responses to the combination therapy, supporting their predictive roles. 
* evaluated the predictive performance of key signature genes of immune cell subsets showing high predictive potentials for the combination therapy (as indicated by both the Pi and immune cell cluster enrichment analyses)
  * selected top-ranked signature genes (defined as marker genes with a score $\ge0.1$ measured using Logistic Regression in the signature gene lists of these immune cell subsets but not in those of other subsets in each major immune compartment
  * CXCL9 and CXCL11 added to the core signature genes of pro-inflammatory macrophages (potential roles in T cell recruitment and ICB response) although not included by our selection procedure
  * evaluated predictive performance of each gene using pre-treatment data of the combination therapy group
    * **scoring classifiers** by the R package ROCR using the clinical outcomes of TNBC patients treated with the combination therapy, in which the clinical response of PR labeled as 1 while SD or PD labeled as 0, and the baseline fractions of immune cells expressing each gene in each major immune compartment. 
    * Signature genes with high AUC values could be explored as potential biomarkers for anti-PD-L1 immunotherapy

---

exhaustion phenotype

TCR sequence

NicheNet analysis

anti-PD-L1 therapy