### The Tumor Profiler Study: Integrated, multi-omic, functional tumor profiling for clinical decision support - Irmisch et al. 2020



a prospective, observational clinical study performed within the **Tumor Profiler (TuPro) Consortium** that aims to show the extent to which such comprehensive information leads to advanced **mechanistic insights** of a patient’s tumor, enables prognostic and predictive **biomarker discovery**, and has the potential to **support clinical decision** making

melanoma, ovarian carcinoma, and acute myeloid leukemia tumors

targeted NGS panel sequencing and digital pathology, we perform extensive characterization using the following exploratory technologies: single-cell genomics and transcriptomics, proteotyping, CyTOF, imaging CyTOF, pharmacoscopy, and 4i drug response profiling (4i DRP)

since NGS-based approaches typically only analyze genomic alterations from bulk tissue, many clinically-relevant questions about the tumor, such as its micro-environment, heterogeneity, and many others, remain unanswered (Friedman et al. 2015) . 



In this study, we explore to which extent a broader and more in-depth functional profiling of tumors can lead to an **enhanced understanding** of the biology of a tumor and whether this new knowledge is **actionable for the treatment** of cancer patients

the Tumor Profiler (TuPro) consortium was formed as a collaborative effort to combine exploratory technologies

. The goal is to decipher the in-depth molecular and functional landscape of patient tumors across different indications, namely melanoma, ovarian carcinoma, and acute myeloid leukemia (AML), to set up a platform for **multi-omics clinical decision support** in an observational clinical study

changes at the DNA, RNA, and protein level are assessed at single-cell resolution in order to characterize both the tumor and the immune compartment. Samples

Samples are also analyzed at the whole-tissue level with targeted NGS, proteotyping, and bulk RNA sequencing

ex vivo drug response testing is performed to assess the effect of selected drugs on cancer cells. Digital

Digital pathology based on large-scale immunohistochemistry analysis, as well as in-depth highly multiplexed imaging analysis, is included to provide spatially resolved data. These

Initial TuPro results show that

* a clinically relevant **turnaround time of two weeks** is feasible for all the technologies included in the trial, 
* analysis results are consistent between technologies
* integrated results provide treatment decision support beyond an approved diagnostic NGS panel sequencing

The goal is to ultimately assess whether a combination of multi- omics and functional profiling techniques (see "Technologies") on the same tumor tissue can supplement emerging standards, such as digital pathology and targeted NGS, to support molecularly informed clinical decision making in tumor boards

The TuPro study aims to determine how each technology contributes to the identification of clinically relevant features in a sample and how the integration of all data types could potentially improve treatment decisions for each patient enrolled in the trial.

**the synergistic effect of this multi-modal approach is expected to enable the establishment of algorithms for integrative analyses and to provide relevant information that can be used to inform patient management directly**

**1cm 3 of solid tumor** material to all the technology platforms, which analyze the sample and return their data within two weeks. 

**three different indications: melanoma, ovarian carcinoma, and acute myeloid leukemia (AML).**

up to **240 patients** will

sequential inclusion of different tumor indications, and clinical studies for all indications will run until the end of 2020

For each indication, the study consists of two parts. 

1. refine and standardize the experimental and analytical processes within the different workstreams
   * data generated by the different technologies are not taken into account when making treatment decisions
   * the results are not discussed with the patients. 
2. molecular summary reports summarizing the findings of the TuPro technologies and the pre-TB treatment recommendations are shared with the Tumor Board
   * The Tumor Board then makes a recommendation based on all the available data. 
   * The final decision on which treatment to implement is recorded.

*Sample preparation and pathology diagnosis*

confirmation of tumor type by conventional pathology 

immunohistochemical staining (IHC) for tumor-specific biomarkers 

assessment of tumor cell content.

We include two emerging clinical diag- nostic approaches, i.e., tests that are not
yet standard in cancer diagnostics, and seven exploratory profiling technologies in the TuPro study (Figures

##### Emerging Standard Diagnostics 

*Digital Pathology*

<u>what ?</u> 

Immune phenotyping 

Expression of the immune checkpoints PD1 and PD-L1 is quantified, and specific features of the immune microenvironment (e.g., the formation of tertiary lymphoid structures) are recorded

Based on the quantity and distribution of the immune markers, tumors are classified as “inflamed” (strong intra- and peritumoral immune cell infiltration), “immune excluded” (immune response around the tumor but no infiltration into tumor cell clusters) or “immune desert” (absence of immune response) to support treatment decisions in conjunction with the TuPro technology nodes.

Digital image analysis is applied to provide unbiased, large-scale quantitative data for the core immune markers on all TuPro samples. This

*Targeted Next-Generation Sequencing*

The assay sequences the complete exons of 324 cancer-related genes for the detection of short variants (single nucleotide variants, insertions, and deletions), copy number alterations (CNAs), as well as microsatellite status and tumor mutational burden. Selected introns and promoter regions of 36 genes are also sequenced for the detection of gene rearrangements and fusions (Frampton

To complement the targeted NGS analyses, we investigate circulating tumor DNA (ctDNA

##### TuPro Exploratory Technologies

*Cell Types, Markers, and Drug Selection* 

Two types of clinically relevant markers are analyzed by TuPro technologies. First, markers that allow a detailed determination of the composition of the tumor and its microenvironment, and second, predictive markers for clinical decision support. These marker sets are selected for each indication separately.

*Single-cell Genomics and Transcriptomics*

droplet-based scDNA-seq and scRNA-seq protocols to identify copy number changes in tumor subclones and their evolutionary relationships, to quantify different cell populations, and to elucidate their gene expression patterns

This approach probes the cellular heterogeneity of the tumor and its local environment and the molecular phenotype in an untargeted and genome-wide fashion on the level of individual cells. Through modular and flexible computational data analysis pipelines, it provides a detailed picture of the tumor and its microenvironment (Tirosh et al. 2016) , as well as its phylogenetic structure (Navin

reveal the distribution of tumor marker genes, drug targets, and immunoregulatory markers across cells

Gene expression profiles of subclones inform on aberrant gene sets and are analyzed further for pathway enrichment and for subclone-specific in-silico drug predictions. The

The presence of different immune cell types and their expression profiles enlightens the interplay between tumor and non-malignant cells and has direct implications for the effectiveness of immunotherapies

may help in the characterization of the driving forces of tumor evolution and guide therapies to become more successful in treating the whole tumor ecosystem.

*Bulk Transcriptomics*

obtain information about alternative splicing, alternative promoters, gene-fusions, RNA-editing and other RNA aberrations

provides a common basis for cross-comparison and joint analyses with samples from TCGA, ICGC, GTEx and other studies

*Proteotype Analysis*

enables the conversion of a clinical sample into a permanent digital proteotype map.

This digital proteotype map contains information about the state of the proteome at the time of the measurement (“proteotype”), which includes the identity and quantity of peptides/proteins, as well as their post-translational modifications

can be analyzed, re-analyzed, compared, and mined in silico to detect and quantify peptide/protein patterns across large clinical cohorts

Data-Independent-Acquisition Mass Spectrometry (DIA-MS) and targeted mass spectrometry (PRM, parallel reaction monitoring) are used to generate such digital proteotype maps

Phenotyping cells/tissues without the need for antibodies using DIA-MS and PRM provides information about the abundance level of peptides/proteins within cells/tissues which are current clinical drug targets and reveals protein abundance patterns which can be correlated across cohort samples from clinical trials for drug response prediction (

*Mass Cytometry (CyTOF)*

simultaneous quantification of over 50 proteins, phosphorylation sites, and transcripts on the single-cell level based on the use of metal-labeled reporters

high dynamic range, low detection limit, high-throughput capability and relatively low cost, this method has proven to be particularly well-suited for the analysis of highly heterogeneous tumor samples

two 40-antibody panels per indication, designed to characterize the immune and the tumor compartment of each sample, respectively, with a specific emphasis on markers which can be targeted in immuno- and onco-therapies

*Imaging Mass Cytometry (IMC)*

quantification of over 50 markers with subcellular resolution on a single tissue section (Giesen

enables the identification of tissue morphology and structural features such as vessels, stroma, tumor tissue phenotypes, and a variety of immune cell types on selected regions of the different tumor samples.

provides high dimensional single-cell phenotyping as its main readout, thus allowing quantification of the proportions of immune and tumor cell populations. Based

40-antibody panel per indication developed

information on spatial interactions between cell populations with a focus on tumor-immune cell interactions is generated which helps in the assessment of immune infiltration patterns (

also enables the identification of the direct neighbors of a specific cell type, including for instance physical interactions between PD-1 positive T cells and PD-L1 expressing tumor or myeloid cells (Giesen

*Pharmacoscopy*

s high-throughput microscopy to measure ex vivo responses to targeted therapies, chemotherapies, and immunotherapies. The

provides a drug sensitivity ranking for a large number of drugs and combinations using a functional relative toxicity measurement of the cancer cells against healthy cells within the sample. The

used to identify effective therapies for individual patients 

improve the understanding of the cellular and molecular systems that determine drug response variability

*4i Drug Response Profiling (4i DRP)*

innovative multiplexed immunofluorescence protocol, termed 4i (iterative indirect immunofluorescence imaging) to achieve an in-depth molecular characterization of phenotypic patient responses to a broad spectrum of drug categories (

Implemented with large-scale image processing, computer vision, and machine learning, the 4i technology is a modern approach in precision medicine to uncover activation of key proliferation and survival pathways as well as the emergence of drug resistance in tumor biopsies profiled with a dedicated drug library, at single-cell resolution. The

The multivariate analysis of multiplexed regulatory signaling proteins in individual cells and the associated multiscale cellular state (morphology, cell size, cell cycle, etc.) allows to decipher cell-to-cell phenotypic variability to different drug treatments and generate predictive models of drug response.

cell heterogeneity in steady-state or in perturbed state can be predicted by a multivariate set of cell features (Snijder

On a larger cohort of patients, this approach may contribute to the identification of predictive and/or prognostic biomarkers and the development of new targeted therapies.

##### Data Analysis, Reports, and Tumor boards The

The clinical and molecular data are collected, stored, and analyzed in a research data management system (RDMS) developed for this project 

see pict



about 40 patients have been analyzed in the TuPro melanoma study and about 15 patients in the TuPro ovarian study. In the vast majority of cases, the observations reported by the technologies support and complement each other, providing higher confidence in treatment decisions. Furthermore, for most melanoma patients discussed so far in the pre-TB, TuPro exploratory data have provided decision support beyond state-of-the-art and emerging standard clinical diagnostics (histopathology



example with a melanoma case study: 

* the described genetic alterations are known in melanoma but are among the less frequent melanoma drivers with an average frequency of <10%
* data generated by the individual TuPro technologies can be utilized in an integrative manner and, importantly, can provide therapeutic decision support that complements emerging standard diagnostic techniques
* the TuPro exploratory technologies not only provided additional evidence on the functional significance of the genetic variants detected by Targeted NGS but have also contributed further information that has allowed treatment recommendations that were not considered before based solely on genetic data .
* might be particularly valuable for cases where the effect of oncogenic drivers is weak, has pleiotropic effects, where it generally remains challenging to single out the primary driving pathway, or on pre-treated tumors where the tumor-maintenance mechanisms are harder to identify.

built a state-of-the-art profiling platform that captures cellular, molecular, spatial, and functional information from three tumor types and generates molecular research and summary reports for a tumor board presentation. Not

enhanced the results of the routine diagnostics that are typically used to select treatments, but they will also provide a powerful database of single-cell, high-dimensional, and functional information for exploratory analyses including biomarker discovery

The comparative analyses of multi-level datasets between patients who responded and those who did not respond to therapy may allow the identification of potential mechanisms of resistance to standard treatment and may thus allow for the identification of potential novel therapeutic targets, as illustrated in the case study.

cohort will still be relatively small at this point of the study, it will allow for the generation of hypotheses that can be tested in subsequent patient samples as well as in publically available datasets or other reference cohorts

to capture the influence of TuPro results in clinical decision making, we plan to adapt the ESCAT scale (Mateo et al. 2018) routinely used to rank genomic alterations as targets of precision medicine to take into account other types of molecular, cellular, and functional data as measured by TuPro emerging technologies



Single-cell genomics approaches (scRNA [Papalexi and Satija, 2018]and scDNA [Kuipers et al., 2020]) generate a high-resolution map of the tumor microenvironment, characterize tumor cell heterogeneity, establish each tumor’s evolutionary history, and take advantage of insights into cancer genomics and tran- scriptomics acquired over the past de- cades.



bulk (DIA-MS) proteo- typing (Gillet et al., 2012; Xuan et al., 2020) and single-cell CyTOF (Wagner et al., 2019) protein-based analyses not only to expand on and translate transcriptomic observations but also to assess post- translational modifications affecting pro- teins involved in signaling pathways

The characterization of the tumor microenvi- ronment is enriched with spatially resolved approaches: digital pathology and imaging mass cytometry (IMC) (Giesen et al., 2014) enable the characterization of cell-cell in- teractions within the tumor microenviron- ment by providing quantitative, single- cell, and spatially resolved data. This is of particular value for predicting the success of therapies that depend on direct cell-cell interactions, such as immune checkpoint inhibitors.

. To understand how the compre- hensive molecular profile translates into drug sensitivity or resistance, we include two ex vivo, single-cell resolution drug response profiling technologies. Pharma- coscopy (Snijder et al., 2017; Vladimer et al., 2017) focuses on cancer-cell- specific drug effectiveness using cell death as a readout, while 4i (iterative indi- rect immunofluorescence imaging) Drug Response Profiling (Gut et al., 2018) maps the changes in proliferation or survival signaling pathways upon drug treatment, using a multiplexed readout of cancer-relevant molecular markers. Both assays screen a cancer-type-specific set of approved or promising off-label cancer drugs, alone or in combination. Finally,

bulk RNA sequencing and targeted DNA sequencing of the tu- mor and of blood-derived cell-free DNA cfDNA). These well-established cancer- profiling molecular approaches enable the comparison to existing large-scale cohort studies, to leverage their informa- tion for patients included in the TuPro study. The

produce a molecular research report (MRR) for each patient. This report is used in a pre-tumor board (pre-TB), where a multidisciplinary group of physicians generates treatment recom- mendations based on three levels of evidence: level A, standard clinical guide- lines (ESMO clinical guidelines); level B, level A plus emerging clinical diagnostic approaches; and level C, all previous evi- dence levels plus data from TuPro explor- atory technologies. Recommendations

Recommen- dations based on level C, along with a synopsis of the pre-TB discussion, are communicated to the tumor board

MRR is made accessible via an interactive
290 Cancer Cell 39, March 8, 2021
web application that provides an overview of potential treatment suggestions along with the specific evidence supporting each option and facilitates discussions between technology experts and clini- cians in the pre-TB. A

A summary of the MRR and the treatment suggestions from the pre-TB is used as a molecular summary report for supporting treatment decisions at the tumor board

data science and machine-learning algorithms will be leveraged to investigate novel molecular markers associated with drug response, marker expression level as a function of diverse clinical variables, and cell popula- tion distribution as a predictor of treat- ment response, among others. The com- bination of bulk and single-cell data collected from a multitude of molecular signals offers opportunities for the devel- opment of new approaches required for data analysis and integration.





*** CASE STUDY

 metastatic acral-lentiginous melanoma 

progressed on anti-PD-1 therapy, planned for palliative chemotherapy



 Histopathology:  

* immune-excluded tumor 
* PD-1 expression on 80% of T cells 
*  PD-L1 expression on 10% of tumor cells

Targeted NGS

*  low tumor mutational burden (1 Mut/Mb) -

*  loss of NF1 exons 24-29 

* ATM , APC, and NF2 loss-of-function (LoF) mutations

* report:

  -> .  MEK inhibitors targeting the NF1 exons 24-29 loss

  -> and PARP inhibitors due to the ATM LoF mutation

  -> . Immunotherapy was not recommended because of the low tumor mutational burden.

pre-TB reocmmendation

* failed immunotherapy, an immune excluded tumor, low tumor mutational burden, and no apparent targets for small molecule therapy

  -> should undergo palliative chemotherapy as planned before the evaluation by the pre-TB



TuPro exploratory technologies CyTOF



CyTOF and Single-cell Genomics & Transcriptomics

* sample composition: 
  * ~35% tumor cells (in at least 4 clusters), 
  * 40% mostly exhausted CD8+ T cells, 
  * 12% macrophages, 
  * the remaining: B lymphocytes and endothelial cells
*  expression of CTLA-4 and PD-1 on T cells 

IMC

*  interaction of PD-1 positive CD8 T cells with PD-L1 positive tumor cells 

  -> potentially supporting checkpoint inhibitor treatment. 

CyTOF and scRNA-seq 

* melanoma population to be positive for melanocytic phenotype markers such as MITF, MELA, and TYRP1, as well as for Vimentin, indicating a transitional phenotype with mesenchymal features 

CyTOF

* very high beta-catenin signal in the main tumor cluster (consistent with the LoF mutation in APC)
*  detect pH2AX expression in a small fraction of tumor cells (potentially corresponding to double-strand breaks caused by  ATM mutation)

previous knowledge

* Genetic alterations in ATM are frequently associated with impaired homologous recombination qualifying for PARP inhibitor treatment

 pharmacoscopy

* no significant cell killing in the presence of PARP inhibitors

 4i DRP 

* no increased apoptosis  in the presence of PARP inhibitors

scRNA-seq

* loss of NF1 expression (supports the loss of NF1 exons 24-29)

4i DRP

* MEK inhibitor treatment with Trametinib significantly reduced pERK expression  and induced apoptosis

pharmacoscopy

* but no significant cell death detected upon Trametinib

previous knowledge

*  Genetic loss of function alterations in NF1 are activators of the MAPK and PI3K/AKT pathways ->  in line with the observed low sensitivity to single MEK inhibitor treatment. 



CyTOF 

*  low pERK signal in tumor cells 

scRNA-seq

* no overactivation of the MAPK pathway 

 -> in this case, MAPK pathway signaling might not be the main driver of tumorigenesis.

Proteotype

*  high CDK4 protein expression  in the tumor

scRNA 

* consistent expression of CDK4 mRNA in the tumor clusters

scDNA-seq

*  low-level copy number gain (CN=3) in a large chromosomal region containing CDK4 (not reported by bulk targeted NGS; below clinically relevance threshold)

previous knowledge

* ATM has also been described in the negative regulation of Cyclins 
*  ATM LoF  might have contributed to the observed increased Cyclin D1 levels, potentially resulting in an activation of the CDK4/6 pathway and CDK4/6 inhibitor sensitivity.





4i DRP

* detected sensitivity to the CDK4/6 inhibitors Ribociclib or Palbociclib , i.e. increased apoptosis and significant oncogenic pathway downregulation 

pharmacoscopy

* detected sensitivity to the CDK4/6 inhibitors: increased melanoma cell death

previous knowledge

*   In addition to targeting cell proliferation, CDK4/6 inhibitors may increase tumor immunogenicity (Goel et al. 2017) . This effect is of particular interest in an immune excluded tumor



pre-TB treatment recommendation 

* off-label CDK4/6 inhibitor in combination with approved anti-PD-1 plus anti-CTLA-4 therapies